### PR TITLE
Make rspec-rails compatible with Rails v6.1.0.rc2

### PR DESCRIPTION
**Pull request by benoittgt**
This PR add:

* Rails v6.1.0.rc2 to the CI build
* Fix 3 errors encountered when running the CI

Answer: https://github.com/rspec/rspec-rails/pull/2398#issuecomment-737757306

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,9 +16,8 @@ jobs:
       fail-fast: false
       matrix:
        include:
-         # Rails 6.1 builds >= 2.5
+         # Rails Master builds >= 2.5
          - ruby: ruby-3.0.0-preview1
-           allow_failure: true
```
Comment (JonRowe) [inline .github/workflows/ci.yml:21]: If we're adding Rails master builds as well as rc builds master should be allowed to fail.

File: example_app_generator/generate_action_mailer_specs.rb [hunk 2]
```diff
@@ -14,15 +14,10 @@
...
-  gsub_file 'config/initializers/action_mailer.rb',
-            /ExampleApp/,
-            Rails.application.class.parent.to_s
 
   copy_file 'spec/support/default_preview_path'
   chmod 'spec/support/default_preview_path', 0755
-  gsub_file 'spec/support/default_preview_path',
-            /ExampleApp/,
-            Rails.application.class.parent.to_s
+
```
Comment (JonRowe) [inline example_app_generator/generate_action_mailer_specs.rb:20]: I'm convinced these are still needed, its just the Rails 6.1 implementation might need to change?

Comment (benoittgt) [reply example_app_generator/generate_action_mailer_specs.rb:20]: I deep dive a little bit on this few weeks ago. The `Rails.application.class.parent` on previous Rails is `ExampleApp`.  So I didn't understand why the gsub was needed.
Also if the CI pass isn't what we want?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (benoittgt) [reply .github/workflows/ci.yml:21]: Ok

**Comment by JonRowe** [comment]
Closing as this has been mostly merged.

**Pull request closed without merging**